keras/callbacks.py

```diff
@@ -779,32 +785,46 @@ def is_indexed_slices(grad):
         else:
             self.writer = tf.summary.FileWriter(self.log_dir)
 
-        if self.embeddings_freq:
+        if self.embeddings_freq and self.embeddings_data is not None:
             embeddings_layer_names = self.embeddings_layer_names
 
             if not embeddings_layer_names:
                 embeddings_layer_names = [layer.name for layer in self.model.layers
                                           if type(layer).__name__ == 'Embedding']
+            self.embeddings = []
+            embeddings_vars = {}
 
-            embeddings = {layer.name: layer.weights[0]
-                          for layer in self.model.layers
-                          if layer.name in embeddings_layer_names}
+            for layer in self.model.layers:
+                if layer.name in embeddings_layer_names:
+
+                    self.batch_id = batch_id = tf.placeholder(tf.int32)
+                    self.step = step = tf.placeholder(tf.int32)
 
-            self.saver = tf.train.Saver(list(embeddings.values()))
+                    embedding_input = self.model.get_layer(layer.name).output
+                    embedding_size = reduce(mul, embedding_input.shape[1:])
```

Comment:
Can `np.prod` be used in this case?

Reply:
Yes, would you prefer that?

Reply:
it's less pythonic but more in line with the style of how things are computed at some other places in the code base.

Reply:
Okay. I thought of using standard lib when possible but I also get that it's good practice to be in line with the rest of the code base. `np.prod` it is, then.